File: socketd/src/main/java/org/noear/socketd/core/Frame.java
package org.noear.socketd.core;

/**
 * 帧
 *
 * @author noear
 * @since 2.0
 */
public class Frame {
    private Flag flag;
    private Message message;

    public Frame(Flag flag, Message message) {
        this.flag = flag;
        this.message = message;
    }

    /**
     * 标志
     * */
    public Flag getFlag(){
        return flag;
    }

    /**
     * 载体
     * */
    public Message getMessage() {
        return message;
    }

    @Override
    public String toString() {
        return "Frame{" +
                "flag=" + flag +
                ", message=" + message +
                '}';
    }
}

File: javascript/ts/org/noear/socketd/transport/core/Frame.ts
import {MessageInternal} from "./MessageInternal";

/**
 * 帧（帧[消息[实体]]）
 *
 * @author noear
 * @since 2.0
 * @param {org.noear.socketd.transport.core.Flag} flag
 * @param {*} message
 * @class
 */
export class Frame {
    /*private*/
    flag: number;

    /*private*/
    message: MessageInternal;

    public constructor(flag: number, message: MessageInternal) {
        this.flag = flag;
        this.message = message;
    }

    /**
     * 标志
     *
     * @return number
     */
    public getFlag(): number {
        return this.flag;
    }

    /**
     * 消息
     *
     * @return {*}
     */
    public getMessage(): MessageInternal {
        return this.message;
    }

    /**
     *
     * @return {string}
     */
    public toString(): string {
        return "Frame{flag=" + this.flag + ", message=" + this.message + '}';
    }
}